test/resource-limits/resource_limits.td

```diff
@@ -0,0 +1,230 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ postgres-connect name=mz_system url=postgres://mz_system:materialize@${testdrive.materialize-sql-addr-internal}
+
+! ALTER SYSTEM SET max_tables TO 42
+contains:unauthorized: only user 'mz_system' is allowed to execute 'ALTER SYSTEM ...'
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_tables = 2
+
+> SHOW max_tables
+2
+
+> CREATE TABLE t1 (a INT)
+
+> CREATE TABLE t2 (a INT)
+
+! CREATE TABLE t3 (a INT)
```

Comment:
Item No 1 .  A check is needed here that using DROP TABLE causes the count to go below the limit , allowing for CREATE TABLE to succeed.

Reply:
Done